src/custom-element.js

```diff
@@ -641,10 +628,28 @@ export function createAmpElementProto(win, name, opt_implementationClass) {
     }
     if (!this.everAttached) {
       this.everAttached = true;
+      if (!isUpgraded(this)) {
+        this.classList.add('amp-unresolved');
+        this.classList.add('-amp-unresolved');
+      }
       try {
-        this.firstAttachedCallback_();
+        this.layout_ = applyLayout_(this);
+        if (this.layout_ != Layout.NODISPLAY &&
+            !this.implementation_.isLayoutSupported(this.layout_)) {
+          throw new Error('Layout not supported for: ' + this.layout_);
+        }
+        this.implementation_.layout_ = this.layout_;
+        this.implementation_.firstAttachedCallback();
       } catch (e) {
         reportError(e, this);
```

Comment:
A nice side effect of the new code: this would have double reported an error in `#firstAttachedCallback`.